modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1438,120 +1438,112 @@ def rebalance_partitions(cls, partitions):
         list[int] or None
             Row lengths if possible to compute it.
         """
+        # We rebalance when the ratio of the number of existing partitions to
+        # the ideal number of partitions is larger than this threshold. The
+        # threshold is a heuristic that may need to be tuned for performance.
+        max_excess_of_num_partitions = 1.5
+        num_existing_partitions = partitions.shape[0]
+        ideal_num_new_partitions = NPartitions.get()
         if (
-            Engine.get() in ["Ray", "Dask", "Unidist"]
-            and StorageFormat.get() == "Pandas"
+            num_existing_partitions
+            <= ideal_num_new_partitions * max_excess_of_num_partitions
         ):
-            # Rebalancing partitions is currently only implemented for PandasOnRay and PandasOnDask.
```

Comment:
All the changes after that are just because of shifting the code by one indent?

Reply:
yes, there were no changes in the logic of this function